model/event/subscribers.go

```diff
@@ -0,0 +1,68 @@
+package event
+
+import (
+	"net/url"
+
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const (
+	ResourceTypeNotification = "notification"
+)
+
+var (
+	subscriberTypeKey   = bsonutil.MustHaveTag(Subscriber{}, "Type")
+	subscriberTargetKey = bsonutil.MustHaveTag(Subscriber{}, "Target")
+)
+
+type Subscriber struct {
+	Type string `bson:"type"`
+	// sad violin
+	Target interface{} `bson:"target"`
+}
+
+func (s *Subscriber) SetBSON(raw bson.Raw) error {
+	temp := struct {
+		Type   string `bson:"type"`
+		Target []byte `bson:"target"`
+	}{}
+	if err := raw.Unmarshal(&temp); err != nil {
+		return errors.Wrap(err, "can't unmarshal subscriber data")
+	}
+	s.Type = temp.Type
+
+	if len(s.Type) == 0 {
+		return errors.New("could not find subscriber type")
+	}
+
+	// TODO registry?
+	switch s.Type {
+	case "github_pull_request":
+		s.Target = githubPullRequestSubscriber{}
+	case "webhook":
+		s.Target = webhookSubscriber{}
+	default:
+		s.Target = string(temp.Target)
```

Comment:
Does it make sense to have this case be here? Do we do something sane if this is a string?

Reply:
Yes, the default case is that the subscriber info can be rendered in just a string, such as emails, slack users, jira projects